### PR TITLE
Empty datafile passed before fetching datafile

### PR DESCRIPTION
**Pull request by Tan108**

### CODE REVIEW

File: src/main/kotlin/com/featurevisor/sdk/Instance.kt [hunk 1]
```diff
@@ -15,6 +16,72 @@ typealias ConfigureBucketValue = (Feature, Context, BucketValue) -> BucketValue
...
+//        val value =  abc.getVariableJSON<Map<String,String>>("live_preroll_ads","live_preroll_watched_duration", getContext())
+//        println("Value : $value")
+//    }else{
+//        println("Not Ready")
+//    }
+
+}
+
+
+private fun getContext(): Map<String, AttributeValue> =
```
Comment (fahad19) [inline src/main/kotlin/com/featurevisor/sdk/Instance.kt:68]: why is this here?

Comment (Tan108) [reply src/main/kotlin/com/featurevisor/sdk/Instance.kt]: It's my mistake. Removed

File: src/main/kotlin/com/featurevisor/sdk/Instance.kt [hunk 2]
```diff
@@ -91,6 +100,7 @@ class FeaturevisorInstance private constructor(options: InstanceOptions) {
                 }
 
                 datafileUrl != null -> {
+                    datafileReader = DatafileReader(options.datafile?: emptyDatafile)
```
Comment (fahad19) [inline src/main/kotlin/com/featurevisor/sdk/Instance.kt:103]: this line means we will use `options.datafile` if it is present, otherwise we fall back to `emptyDatafile`?

Comment (Tan108) [reply src/main/kotlin/com/featurevisor/sdk/Instance.kt:103]: Yes

File: src/main/kotlin/com/featurevisor/sdk/Instance.kt [hunk 3]
```diff
@@ -15,6 +15,15 @@ typealias ConfigureBucketValue = (Feature, Context, BucketValue) -> BucketValue
 typealias InterceptContext = (Context) -> Context
 typealias DatafileFetchHandler = (datafileUrl: String) -> Result<DatafileContent>
 
+var emptyDatafile = DatafileContent(
+    schemaVersion =  "1",
+    revision = "unknown",
+    attributes = emptyList(),
+    segments = emptyList(),
+    features = emptyList(),
+
```
[the remark this answers is not in the document]
Comment (Tan108) [reply src/main/kotlin/com/featurevisor/sdk/Instance.kt]: done
